src/main/java/bio/terra/service/job/StairwayExceptionSerializer.java

```diff
@@ -0,0 +1,114 @@
+package bio.terra.service.job;
+
+import bio.terra.common.exception.DataRepoException;
+import bio.terra.service.job.exception.JobResponseException;
+import bio.terra.service.job.exception.SerDesException;
+import bio.terra.stairway.ExceptionSerializer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+public class StairwayExceptionSerializer implements ExceptionSerializer {
+    private ObjectMapper objectMapper;
+
+    public StairwayExceptionSerializer(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public String serialize(Exception exception) {
+        if (exception == null) {
+            return StringUtils.EMPTY;
+        }
+
+        // Wrap non-runtime exceptions so they can be rethrown later
+        if (!(exception instanceof RuntimeException)) {
+            exception = new JobResponseException(exception.getMessage(), exception);
+        }
+
+        StairwayExceptionFields fields = new StairwayExceptionFields();
+
+        if (exception instanceof DataRepoException) {
+            fields
+                .setDataRepoException(true)
+                .setClassName(exception.getClass().getName())
+                .setMessage(exception.getMessage())
+                .setErrorDetails(((DataRepoException) exception).getErrorDetails());
+        } else {
+            fields
+                .setDataRepoException(false)
+                .setClassName(exception.getClass().getName())
+                .setMessage(exception.getMessage());
+        }
+
+        try {
+            return objectMapper.writeValueAsString(fields);
+        } catch (JsonProcessingException ex) {
+            throw new SerDesException("This should never happen", ex);
```

Comment:
Getting to this catch statement means that we have a RuntimeException (which is possible a checked Exception wrapped in a JobResponseException) that Jackson failed to serialize. Why do you say that this should never happen?

Reply:
Actually, maybe I can answer my own question. This should never happen because you've designed the StairwayExceptionFields class to be simple enough to always be serializable. Is that right? If so, I think it'd be helpful to add that as a comment here and maybe to the exception message also.

Reply:
You are right and I added a comment.